framework/src/interfaces/TaggingInterface.C

```diff
@@ -359,21 +381,73 @@ TaggingInterface::prepareMatrixTagLower(Assembly & assembly,
 void
 TaggingInterface::accumulateTaggedLocalResidual()
 {
-  for (auto & re : _re_blocks)
-    *re += _local_re;
-  for (auto & absre : _absre_blocks)
-    for (const auto i : index_range(_local_re))
-      (*absre)(i) += std::abs(_local_re(i));
+  if (_bx_norm)
+  {
+    DenseVector<Number> scaled_local_re = _local_re;
+    if (_eigen_problem->negativeSignEigenKernel())
+      scaled_local_re.scale(1 / (*_bx_norm));
+    else
+      scaled_local_re.scale(-1 / (*_bx_norm));
```

Comment:
these operations are all unnecessary for the most common case in which you do not have any additional tagged vectors on top of your eigen tag

Reply:
This is inside of a if statement `if (_bx_norm)`, which should not introduce much computing burden. Not sure whether I answered this comment.

Reply:
New implementation should probably resolve this comment. I left the another related comment open and will resolve this.